### DISCUSSION TITLE
Dynamic and Static routes with next.js

### DISCUSSION THREAD

**Discussion by krish-penumarty** [vercel/next.js#29618] [Help] [answered]
Hello,

I had a quick question about file based routing. Curious how to set it up for certain conditions. For example if I had a path such as `/foo/bar` that would map to `/foo/[id].js` in my file based router. 

Is it possible to have `/foo/baz` match `/foo/baz.js` and not have it match `/foo/[id].js. I tested it locally and it works, but want to make sure this is the appropriate way of handling it?

**Comment by ijjk** [comment, marked as answer]
Hi, yes that is correct static pages will match before dynamic pages so that is the appropriate way to handle what you are describing.

**Reply by krish-penumarty** [reply]
Thank you for your answer!
